src/Core/Grid/Query/CountryQueryBuilder.php

```diff
@@ -96,26 +103,20 @@ protected function getQueryBuilder(SearchCriteriaInterface $searchCriteria): Que
     {
         $qb = $this->connection->createQueryBuilder()
             ->from($this->dbPrefix . 'country', 'c')
-            ->innerJoin(
```

Comment:
This query builder is also used by the list, and I guess in the list we want to display only the countries associated with the shop context so this condition seems important

To avoid having duplicate categories you can add a `groupBy('c.id_country')`

Reply:
I see you added the `groupBy` which is good but I suggested this to avoid the bug you mentioned related to multi shop I'm still not convinced that the multishop condition should be removed Did you test your PR in a multishop env and tried to assign/unassign countries to some shops? They should be filtered from the list depending on their association configuration

Reply:
:arrow_up: still seems to be relevant